post-to-aws/src/main/java/no/sikt/nva/brage/migration/aws/S3RecordStorage.java

```diff
@@ -0,0 +1,136 @@
+package no.sikt.nva.brage.migration.aws;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import no.sikt.nva.brage.migration.common.model.record.Record;
+import no.sikt.nva.brage.migration.common.model.record.content.ContentFile;
+import nva.commons.core.SingletonCollector;
+import nva.commons.core.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class S3RecordStorage implements StoreRecord {
+
+    public static final String COULD_NOT_WRITE_MESSAGE = "Could not write files to s3 for: ";
+    public static final String JSON_STRING = ".json";
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String bucketName = "anette-kir-brage-migration-experiment";
+    private static final Logger logger = LoggerFactory.getLogger(S3RecordStorage.class);
+    private final S3Client s3Client;
+    private String pathPrefixString = StringUtils.EMPTY_STRING;
+
+    public S3RecordStorage(S3Client s3Client, String pathPrefixString) {
+        this.s3Client = s3Client;
+        this.pathPrefixString = pathPrefixString;
+    }
+
+    public S3RecordStorage(S3Client s3Client) {
+        this.s3Client = s3Client;
+    }
+
+    @Override
+    public void storeRecord(Record record) {
+        try {
+            writeRecordFilesToS3(record);
```

Comment:
nitpicking: men kan denne kalles for writeAssociatedFilesToS3?

Reply:
godt forslag :)